pydantic/v1/typing.py

```diff
@@ -0,0 +1,83 @@
+# ruff: noqa: F403,F401,F405
+# flake8: noqa: F401,F405
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from pydantic.typing import *
+else:
+    # explicit importing
+    from pydantic.typing import (
```

Comment:
Note I'm *a bit* worried about these, for example if a module uses a `__all__` it does not stop someone using `from pydantic.<module> import <SymbolNotProvidedInAllButPresent>` whereas with these imports `from pydantic.v1.<module> import <SymbolNotProvidedInAllButPresent>` will *not* be found. 

I'm not sure _how common_ this will actually be (I'm sure _some people_ will definitely import objects from a module that does not have them defined in `__all__` 🙃) but I'm also not too sure whether we _want_ to support that in this either. 

Personally I'm _kinda_ fine with this, but happy to take opinions. Also, this will likely need a migration guide update too right?

Reply:
I'd argue that `from pydantic.<module> import <SymbolNotProvidedInAllButPresent>` amounts to depending on an undocumented implementation detail in the old API, and that you should be free to avoid replicating that in the "new" `pydantic.v1.*` API.